docs/src/architecture/notes.md

```diff
@@ -1,73 +1,194 @@
-
-# Note storage modes
-Similar to accounts, there are two storage modes for notes in Miden. Notes can be stored privately in the [Notes DB](https://0xpolygonmiden.github.io/miden-base/architecture/state.html#notes-database) with only the note hash. Or notes can be stored publicly with all data.
-
-Privately stored notes can only be consumed if the note data is known to the consumer. That means, there must be some offchain communication to transmit the note's data from the sender to the receipient.
-
-# Note metadata
-For every note the Miden Operator stores metadata in the Note DB. This metadata includes:
-
-* A **user-defined tag** as a means to quickly grab all notes for a certain application or use case.
-* A **sender** to be able to provide also ERC20 contract functionality.
-* The **number of assets** contained in the note. 
-
-# Note hash
-The note hash is computed as:
+ In addition, a note has **metadata** including the sender, tag, and number of assets.
 
-`hash(hash(hash(hash(serial_num, [0; 4]), script_hash), input_hash), vault_hash)`
+# Note's lifecycle
+New notes are being created when executing a transaction. After verifying the transaction proof the Operator adds either only the note hash or the full note data to the [Notes DB](https://0xpolygonmiden.github.io/miden-base/architecture/state.html#notes-database). Notes can be produced and consumed locally by users in local transactions or by the operator in a network transaction. Note consumption requires the transacting party to know the note data to compute the nullifier. After successful verification, the Operator sets the corresponding entry in the Nullifier DB to `1`. 
 
-This achieves the following properties:
-- Every note can be reduced to a single unique hash.
-- To compute a note's hash, we do not need to know the note's `serial_num`. Knowing the hash
-    of the `serial_num` (as well as `script_hash`, `input_hash` and `note_vault`) is sufficient.
-- Moreover, the note hash can be computed having the [recipient](https://0xpolygonmiden.github.io/miden-base/architecture/notes.html#note-recipient) and note vault, as `hash(recipient, vault_hash)`.
-- We compute the hash of `serial_num` as `hash(serial_num, [0; 4])` to simplify processing within the VM.
-
-# Note nullifier
-The nullifier is the note's index in the [Nullifier DB](https://0xpolygonmiden.github.io/miden-base/architecture/state.html#nullifier-database). The Nullifier DB stores the information whether a note was already consumed.
+<p align="center">
+    <img src="../diagrams/architecture/note/Note_life_cycle.png">
+</p>
 
-The nullifier is computed as `hash(serial_num, script_hash, input_hash, vault_hash)`.
+The following section will explain, how notes are created, stored, discovered and consumed. 
+
+## Note creation
+Notes are created in Miden transactions. If a valid Miden transaction outputting an `OutputNote` gets verified by the Miden Operator it gets added to the Notes DB.
```

Comment:
Change to by miden transactions. 

Change Operator to "network" or "rollup"

"Notes DB" can be an impl detail, We could say that it is recorded by the network or that it has been added to the rollup's state.

Reply:
I will try to keep the operator and Note DB. It might be an implementation detail. On the other hand, the three different DBs (e.g., Note DB) are part of the architecture and even protocol. 

We can change that in a future PR.